### PR TITLE
Add support for running with tfm=net9.0

### PR DESCRIPTION
**Pull request by radical**
- ChannelMap: add entry for 9.0
- nuget.config: add dotnet9 source
- Add 9.0 versions for package references

### CODE REVIEW

**Comment by radical** [comment]
Don't merge yet.

**Comment by LoopedBard3** [comment]
@radical the 0.13.10-nightly.20231019.90 version should now be available.

File: scripts/channel_map.py [hunk 1]
```diff
@@ -3,8 +3,8 @@
 class ChannelMap():
     channel_map = {
         'main': {
-            'tfm': 'net8.0',
-            'branch': '8.0',
+            'tfm': 'net9.0',
+            'branch': '9.0',
             'quality': 'daily'
         },
         '8.0': {
```
Comment (LoopedBard3) [inline scripts/channel_map.py:10]: We should add 9.0 also to the channel_map, though that can be done in another PR.

**Review by LoopedBard3 — APPROVED** [review]
LGTM

File: NuGet.config [hunk 2]
```diff
@@ -6,6 +6,7 @@
   <packageSources>
     <clear />
     <add key="dotnet-public" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public/nuget/v3/index.json" />
+    <add key="dotnet9" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet9/nuget/v3/index.json" />
```
Comment (lewing) [inline NuGet.config:9]:
```suggestion
    <add key="dotnet9" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet9/nuget/v3/index.json" />
    <add key="dotnet9-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet9-transport/nuget/v3/index.json" />
```
Not sure if it is required but it shouldn't hurt and dotnet8-transport is listed

File: scripts/channel_map.py (hunk 1 above)
Comment (radical) [reply scripts/channel_map.py:10]: done.

**Comment by radical** [comment]
Failure is relevant, investigating now.

File: scripts/channel_map.py [hunk 3]
```diff
@@ -3,8 +3,13 @@
 class ChannelMap():
     channel_map = {
         'main': {
-            'tfm': 'net8.0',
-            'branch': '8.0',
+            'tfm': 'net9.0',
```
Comment (DrewScoggins) [inline scripts/channel_map.py:6]: Should we also go ahead and add all of the release/9.0* branches as well?

**Comment by radical** [comment]
The blazor_scenarios fails with:
`[2023/10/19 18:04:43][INFO] /datadisks/disk1/work/BA02099F/p/artifacts/packages/microsoft.net.sdk.webassembly.pack/9.0.0-alpha.1.23470.17/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets(157,38): error MSB4057: The target "WasmBuildApp" does not exist in the project. [/datadisks/disk1/work/BA02099F/w/AFEA0949/e/app/BlazingPizza.Client/BlazingPizza.Client.csproj::TargetFramework=net9.0]`

.. which is because the latest sdk still has a much older `WorkloadManifest` (note `9.0.0-alpha.1.23470.17` in the path) which is missing bunch of changes needed to support building with net9.0, thus the webassembly.sdk is not imported and we get this error.
This will have to be ignored for now, as the the flow of `runtime` through to `installer` is stuck right now. But we can validate the changes with a run on `dotnet-runtime-perf`.

The `sdk_scenarios` failures are not related, IIUC.

**Comment by radical** [comment]
A new installer is available, so I triggered a rebuild.

**Comment by radical** [comment]
All the blazor_scenarios pass now with the new installer! Only unrelated failures are left.

**Pull request merged**